compiler/AST/symbol.cpp

```diff
@@ -2332,3 +2332,21 @@ const char* toString(Symbol* sym, bool withType) {
 
   return sym->name;
 }
+
+struct SymbolPairComparator {
+  bool operator()(std::pair<Symbol*, Symbol*> lhs,
+                  std::pair<Symbol*, Symbol*> rhs) {
+    std::less<Symbol*> lessSym;
+
+    return lessSym(lhs.first, rhs.first);
+  }
+};
+
+SymbolMapVector sortedSymbolMapElts(const SymbolMap& map) {
+  std::vector<std::pair<Symbol*, Symbol*> > elts;
```

Comment:
Consider resizing `elts` upfront to `map.n/2` or another appropriate size.

Reply:
I don't think it's so easy to see the number of keys in the map and I'm not sure of the value of guessing the size. This seems like something to consider only for performance implications and I'm not measuring performance in this effort.